### PR TITLE
Add harbor package 2.3.3

### LINKED ISSUE #2412: Harbor package 2.3.3

### PR DESCRIPTION
**Pull request by zyyw**
Signed-off-by: Shengwen Yu <yshengwen@vmware.com>

## What this PR does / why we need it

Add Harbor package 2.3.3

## Details for the Release Notes (PLEASE PROVIDE)

```release-note
The Harbor package has been updated to version 2.3.3.
```

## Which issue(s) this PR fixes

Fixes: #2412  

## Describe testing done for PR

`cd addons/packages/harbor/2.3.3/test`
`make test`

### CODE REVIEW

**Comment by stmcginnis** [comment]
Updated the title and description to include release note. Please make sure to include good descriptions of the change to all issues and PRs. Thanks!

File: addons/packages/harbor/2.3.3/README.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
+# harbor Package
```
Comment (stmcginnis) [inline addons/packages/harbor/2.3.3/README.md:1]: s/harbor/Harbor/

File: addons/packages/harbor/2.3.3/README.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+This package provides << awesome functionality >> using [harbor](https://INFO_NEEDED).
```
Comment (stmcginnis) [inline addons/packages/harbor/2.3.3/README.md:3]: Please fill in placeholder explaining what the package provides.

File: addons/packages/harbor/2.3.3/README.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+The following configuration values can be set to customize the harbor installation.
+
+### Global
+
+| Value | Required/Optional | Description |
+|-------|-------------------|-------------|
+| `name` | Optional | description |
+
+### harbor Configuration
```
Comment (stmcginnis) [inline addons/packages/harbor/2.3.3/README.md:17]: s/harbor/Harbor/

File: addons/packages/harbor/2.3.3/README.md [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+| Value | Required/Optional | Description |
+|-------|-------------------|-------------|
+| `name` | Optional | description |
+
+## Usage Example
+
+The follow is a basic guide for getting started with harbor.
```
Comment (stmcginnis) [inline addons/packages/harbor/2.3.3/README.md:25]: Please complete this section or remove.

**Review by stmcginnis — CHANGES_REQUESTED** [review]

**Review by stmcginnis — DISMISSED** [review]
Updates look good to me, thanks!

**Review by heww — APPROVED** [review]

**Review by karuppiah7890 — APPROVED** [review]
I only checked `test/e2e/addons_config.yaml` from an approval (file ownership) point of view. I'll look at whole PR soon

LGTM!

**Review by MinerYang — APPROVED** [review]
LGTM

**Comment by heww** [comment]
@seemiller Could you help to review this PR? Seems that the `test/e2e/addons_config.yaml` or `addons/repos/main.yaml` file is modified, so it requires the tce-owners to review this PR. Thanks!

**Review by stmcginnis — APPROVED** [review]
I think with the approval from @MinerYang we should be good to go.

**Pull request merged**